src/plan/mutator_context.rs

```diff
@@ -481,8 +530,18 @@ pub(crate) fn create_allocator_mapping(
     if include_common_plan {
         map[AllocationSemantics::Immortal] = reserved.add_bump_pointer_allocator();
         map[AllocationSemantics::Los] = reserved.add_large_object_allocator();
-        // TODO: This should be freelist allocator once we use marksweep for nonmoving space.
-        map[AllocationSemantics::NonMoving] = reserved.add_bump_pointer_allocator();
+        map[AllocationSemantics::NonMoving] = if cfg!(not(any(
+            feature = "immortal_as_nonmoving",
+            feature = "marksweep_as_nonmoving"
+        ))) {
+            reserved.add_immix_allocator()
+        } else if cfg!(feature = "marksweep_as_nonmoving") {
+            reserved.add_free_list_allocator()
+        } else if cfg!(feature = "immortal_as_nonmoving") {
+            reserved.add_bump_pointer_allocator()
+        } else {
+            panic!("No policy selected for nonmoving space")
```

Comment:
Actually the first branch `cfg!(not(any(feature = "immortal_as_nonmoving", feature = "marksweep_as_nonmoving")))` covers all other cases, so this `else` is unreachable.  But I suggest we move `reserved.add_immix_allocator()` into this `else` branch and removt the first `if cfg!(not(any(..., ...)))` branch.